ee/tabby-webserver/src/service/web_documents.rs

```diff
@@ -1,21 +1,42 @@
-use std::sync::Arc;
+use std::{
+    collections::{HashMap, HashSet},
+    sync::Arc,
+};
 
+use anyhow::anyhow;
 use async_trait::async_trait;
 use juniper::ID;
-use tabby_db::DbConn;
+use tabby_db::{DbConn, WebDocumentDAO};
 use tabby_schema::{
-    job::JobService,
+    job::{JobInfo, JobService},
     web_documents::{CustomWebDocument, PresetWebDocument, WebDocumentService},
-    Result,
+    AsID, AsRowid, CoreError, Result,
+};
+
+use super::{
+    background_job::BackgroundJobEvent, graphql_pagination_to_filter,
+    preset_web_documents_data::PRESET_WEB_DOCUMENTS_DATA,
 };
 
 pub fn create(db: DbConn, job_service: Arc<dyn JobService>) -> impl WebDocumentService {
-    WebDocumentServiceImpl { db, job_service }
+    let data: serde_json::Value = serde_json::from_str(PRESET_WEB_DOCUMENTS_DATA).unwrap();
+    let mut preset_web_documents = HashMap::new();
+    for doc in data.as_array().unwrap() {
+        let name = doc.get("name").unwrap();
+        let url = doc.get("crawlerStart").unwrap();
+        preset_web_documents.insert(name.to_string(), url.to_string());
+    }
```

Comment:
```suggestion
```

In our previous discussion, we've concluded that inactive data shall not being inserted to db.